Archive/Imo/Imo1982Q3.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios, Alex Brodbelt
+-/
+import Mathlib.Algebra.Order.BigOperators.Ring.Finset
+import Mathlib.Algebra.GeomSum
+import Mathlib.Data.NNReal.Basic
+
+/-!
+# IMO 1982 Q3
+
+Consider infinite sequences $\{x_n\}$ of positive reals such that $x_0 = 1$ and
+$x_0 \ge x_1 \ge x_2 \ge \ldots$
+
+a) Prove that for every such sequence there is an $n \ge 1$ such that:
+
+$$\frac{x_0^2}{x_1} + \ldots + \frac{x_{n-1}^2}{x_n} \ge 3.999$$
+
+b) Find such a sequence such that for all $n$:
+
+$$\frac{x_0^2}{x_1} + \ldots + \frac{x_{n-1}^2}{x_n} < 4$$
+
+The solution is based on Solution 1 from the
+[Art of Problem Solving](https://artofproblemsolving.com/wiki/index.php/1982_IMO_Problems/Problem_3)
+website. For part a, we use Sedrakyan's lemma to show the sum is bounded below by
+$\frac{4n}{n + 1}$, which can be made arbitrarily close to $4$ by taking large $n$. For part b, we
+show the sequence $x_n = 2^{-n}$ satisfies the desired inequality.
+-/
+
+open Finset NNReal
+
+variable {x : ℕ → ℝ} {n : ℕ} (hn : n ≠ 0) (hx : Antitone x)
+
+namespace Imo1982Q3
+include hn hx
+
+/-- `x (n + 1)` is at most the average of `x 1`, `x 2`, ..., `x n`. -/
```

Comment:
This docstring does not seem to correspond to the lemma it's documenting.

Reply:
The actual lemma is just this plus `∑ k ∈ range n, x k` added to both sides.

Reply:
That's not true either because of various errors involving bounds (`range n` is 0<i<n, right?).

Reply:
You're right, I've updated the docstring.

Reply:
So now the docstring does claim a result which is mathematically equivalent to the one being proved, but it still feels very weird to me that the docstring doesn't actually say what the theorem says. Everything else in this PR looks fine but can't you at least explain in the docstring that the theorem doesn't actually say what you say it says? i.e. incorporate your comment above about what the lemma actually says, into the docstring? Or just actually prove the result you claim in the docstring as another lemma? Why prove X and then claim that you're proving X'? It just seems illogical to me.

Reply:
I've added my comment on `∑ k ∈ range n, x k` to the docstring.